### DISCUSSION TITLE
Перезагрузка ESP32

### DISCUSSION THREAD

**Discussion by HuskySmile** [vvip-68/LedPanelWiFi#14] [Вопросы и ответы] [answered]
![Скриншот-06-11-2023 23_56_48](https://github.com/vvip-68/LedPanelWiFi/assets/96444332/a432377f-8d79-4f3c-91cb-91b696737372)
Добрый день. Собрал матрицу 43х35, все прошилось без ошибок. Подключаюсь с PC и в веб интерфейсе все работает, но стоит попробовать сменить эффект плата тут-же перезагружается и получается нет возможности настроить матрицу. 
Помогите пожалуйста, как это исправить если возможно?

**Comment by vvip-68** [comment]
По этому кусочку лога ничего не понятно, никаких выводов сделать невозможно.
По надписи CORRUPT HEAP можно судить, что нарушено распределение данных в оперативной памяти.
Но вот программная эта ошибка из за нехватки памяти или аппаратная - может "битая память" - непонятно.
Нехватка памяти - по мне так маловероятна - чуть более 1500 диодов. Свободной памяти - достаточно - более 200К всего, максимальный кусок - 112К... Более чем достаточно.

Я "гонял" тесты с 16 тыс диодов - вылетов не было.
Это при переключении на определенный режим или на любой такое происходит?
Есть ли возможность не переключая эффект в настройках матрицы в WebUI указать меньший размер матрицы?
Понятно, что изображение "собьется", но просто для теста - как будет вести себя с меньшим количеством диодов.

Есть ли другой экземпляр микроконтроллера просто попробовать на нем?
У вас в логе указан режим работы память - mode: DIO
Режим по умолчанию - QIO. Проверьте в настройках в меню "Инструменты" какой режим там стоит.

**Reply by HuskySmile** [reply]
Добрый вечер, все заработало!!!
Как это пофиксилось не знаю. Я просто еще пару раз с нуля все прошил, менял разные Ip. Но как мне кажется, я что-то напутал с загрузкой данных в файловую систему МК.

> В папке скетча создайте подпапку  [data](https://github.com/vvip-68/LedPanelWiFi/wiki/%D0%97%D0%B0%D0%B3%D1%80%D1%83%D0%B7%D0%BA%D0%B0-%D0%B4%D0%B0%D0%BD%D0%BD%D1%8B%D1%85-%D0%B2-%D1%84%D0%B0%D0%B9%D0%BB%D0%BE%D0%B2%D1%83%D1%8E-%D1%81%D0%B8%D1%81%D1%82%D0%B5%D0%BC%D1%83-%D0%9C%D0%9A#%D0%B2%D0%BD%D0%B8%D0%BC%D0%B0%D0%BD%D0%B8%D0%B5)

Я не мог понять где конкретно нужно создать эту папку, поскольку думал что это и есть папка скетча, но в ней уже есть такая папка.
![Скриншот-06-11-2023 23_56_48](https://github.com/vvip-68/LedPanelWiFi/assets/96444332/1ba6a806-ec29-43ba-bd1f-05fd69201802)
В итоге я создал ее тут и в ней еще папку 43p35
![Скриншот-07-11-2023 18_18_38](https://github.com/vvip-68/LedPanelWiFi/assets/96444332/eeab215b-bdd8-4964-a687-e6055f0ce3c5)
После этого нажал "ESP32 LittleFS Data Upload", дождался загрузки и все заработало. Но это все догадки, истиной причины я не знаю поскольку я ни бум бум в программирование и подобном. Если добавите пару скринов в описание про загрузку в МК думаю будет не лишним для таких как я и подобным))) Спасибо огромное что развиваете и не бросаете проект!!!

> Это при переключении на определенный режим или на любой такое происходит?

Это происходило при попытке смены любого эффекта, но интересно то что другой функционал работал к примеру изменение яркости

> Есть ли возможность не переключая эффект в настройках матрицы в WebUI указать меньший размер матрицы?

Да, пробовал разные размеры, но никак не помогало. Все равно перезагружался...

> Есть ли другой экземпляр микроконтроллера просто попробовать на нем?

К сожалению нет, уже мысли появились на 8266 пробовать

> У вас в логе указан режим работы память - mode: DIO
Режим по умолчанию - QIO. Проверьте в настройках в меню "Инструменты" какой режим там стоит.

Перепроверял, но все равно в логе пишет mode: DIO

**Reply by vvip-68** [reply]
> Я не мог понять где конкретно нужно создать эту папку, поскольку думал что это и есть папка скетча, но в ней уже есть такая папка.

Ну раз есть - ничего создавать не нужно. Это она и есть.

> В итоге я создал ее тут и в ней еще папку 43p35

Нет, тут это ничего не дает. 

> Перепроверял, но все равно в логе пишет mode: DIO

Ну тут я на всякий случай спросил. Итак, причина так и осталась невыясненной.

**Reply by vvip-68** [reply, marked as answer]
А ответ то давно вот тут описан. Сам на это напоролся и вот только тогда вспомнил:


![image](https://github.com/vvip-68/LedPanelWiFi/assets/39353139/ed71017b-77d5-4504-926f-a1b0eb578ac4)
